packages/patternfly-4/react-core/src/components/List/List.tsx

```diff
@@ -1,27 +1,50 @@
 import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/List/list';
 import { css, getModifier } from '@patternfly/react-styles';
+import { Omit } from '../../helpers/typeUtils'
+
+export enum OrderType {
+  number = '1',
+  lowercaseLetter = 'a',
+  uppercaseLetter = 'A',
+  lowercaseRomanNumber = 'i',
+  uppercaseRomanNumber = 'I',
+}
 
 export enum ListVariant {
-  inline = 'inline'
+  inline = 'inline',
+  ordered = 'ol',
+  unordered = 'ul',
 }
 
-export interface ListProps extends React.HTMLProps<HTMLUListElement> {
+export interface ListProps extends Omit<React.HTMLProps<HTMLUListElement | HTMLOListElement>, 'type'> {
   /** Anything that can be rendered inside of the list */
   children?: React.ReactNode;
-  /** Additional classes added to the list. */
+  /** Additional classes added to the list */
   className?: string;
   /** Adds list variant styles */
-  variant?: 'inline';
+  variant?: ListVariant.inline;
+  /** Sets the way items are numbered if variant is set to ordered */
+  type?: OrderType;
+  component?: ListVariant.ordered | ListVariant.unordered;
 }
 
 export const List: React.FunctionComponent<ListProps> = ({
   className = '',
   children = null,
   variant = null,
+  type = OrderType.number,
+  ref = null,
+  component = ListVariant.unordered,
```

Comment:
using `ListVariant` for the `component` prop type ?

Reply:
sorry, didn't notice this is done in Text component too, ok by me.

Reply:
I can change that to `ListComponent` if you like, unless you have a better name.
I am happy to hear any idea. :D

Reply:
I like the `isSomething` pattern:
for example: leave the variant like it was in the original component, and add a `isOrdered` prop ...

But since I saw that `Text` is using the `TextVariant` for it's `component` view, I'm fine with continuing the same pattern.